virttest/utils_disk.py

```diff
@@ -182,6 +190,571 @@ def clean_old_image(image):
         os.remove(image)
 
 
+def get_linux_disks(session, partition=False):
+    """
+    List all disks or disks with no partition.
+
+    :param session: session object to guest
+    :param partition: if true, list all disks; otherwise,
+                      list only disks with no partition.
+    :return: the disks dict.
+             e.g. {kname: [kname, size, type, serial, wwn]}
+    """
+    list_disk_cmd = "lsblk -o KNAME,SIZE,TYPE,SERIAL,WWN"
+    get_disk_name_cmd = "lsblk -no pkname /dev/%s"
+    output = session.cmd_output(list_disk_cmd)
+    devs = output.splitlines()
+    disks_dict = {}
+    part_dict = {}
+    for dev in devs:
+        dev = dev.split()
+        if "disk" in dev:
+            disks_dict[dev[0]] = dev
+        if "part" in dev:
+            part_dict[dev[0]] = dev
+    if partition:
+        disks_dict = dict(disks_dict, **part_dict)
+    else:
+        for part in part_dict.keys():
+            output = session.cmd_output(get_disk_name_cmd % part)
+            disk = output.splitlines()[0].strip()
+            if disk in disks_dict.keys():
+                disks_dict.pop(disk)
+    return disks_dict
+
+
+def get_windows_disks_index(session, image_size):
+    """
+    Get all disks index which show in 'diskpart list disk'.
+    except for system disk.
+    in diskpart: if disk size < 8GB: it displays as MB
+                 else: it displays as GB
+
+    :param session: session object to guest.
+    :param image_size: image size. e.g. 40M
+    :return: a list with all disks index except for system disk.
+    """
+    disk = "disk_" + ''.join(random.sample(string.ascii_letters + string.digits, 4))
+    disk_indexs = []
+    list_disk_cmd = "echo list disk > " + disk
+    list_disk_cmd += " && echo exit >> " + disk
+    list_disk_cmd += " && diskpart /s " + disk
+    list_disk_cmd += " && del /f " + disk
+    disks = session.cmd_output(list_disk_cmd)
+    size_type = image_size[-1] + "B"
+    if size_type == "MB":
+        disk_size = image_size[:-1] + " MB"
+    elif size_type == "GB" and int(image_size[:-1]) < 8:
+        disk_size = str(int(image_size[:-1]) * 1024) + " MB"
+    else:
+        disk_size = image_size[:-1] + " GB"
+    regex_str = 'Disk (\d+).*?%s.*?%s' % (disk_size, disk_size)
+    for disk in disks.splitlines():
+        if disk.startswith("  Disk"):
+            o = re.findall(regex_str, disk, re.I | re.M)
+            if o:
+                disk_indexs.append(o[0])
+    return disk_indexs
+
+
+def _wrap_windows_cmd(cmd):
+    """
+    add header and footer for cmd in order to run it in diskpart tool.
+
+    :param cmd: cmd to be wrapped.
+    :return: wrapped cmd
+    """
+    disk = "disk_" + ''.join(random.sample(string.ascii_letters + string.digits, 4))
+    cmd_header = "echo list disk > " + disk
```

Comment:
Better add `echo rescan > cmd && echo select disk 1 >> cmd && echo clean >> `
May be add a force create partition option.
```
C:\>echo rescan > cmd && echo select disk 1 >> cmd  && echo create partition primary >> cmd && echo select partition 1 >> cmd && echo assign letter=I >> cmd && echo exit >> cmd && diskpart /s cmd

Microsoft DiskPart version 6.1.7601
Copyright (C) 1999-2008 Microsoft Corporation.
On computer: WIN-M02HSVFMND8

Please wait while DiskPart scans your configuration...

DiskPart has finished scanning your configuration.

Disk 1 is now the selected disk.

Virtual Disk Service error:
There is not enough usable space for this operation.


C:\>echo rescan > cmd && echo select disk 1 >> cmd && echo clean >> cmd && echo create partition primary >> cmd && echo select partition 1 >> cmd && echo assign letter=I >> cmd && echo exit >> cmd && diskpart /s cmd

Microsoft DiskPart version 6.1.7601
Copyright (C) 1999-2008 Microsoft Corporation.
On computer: WIN-M02HSVFMND8

Please wait while DiskPart scans your configuration...

DiskPart has finished scanning your configuration.

Disk 1 is now the selected disk.

DiskPart succeeded in cleaning the disk.

DiskPart succeeded in creating the specified partition.

Partition 1 is now the selected partition.

DiskPart successfully assigned the drive letter or mount point.

Leaving DiskPart..
```

Reply:
Thank you in advance.
1. we can get all  empty disks by function \"get_windows_disks_index\" before format disk to avoid force create partition. 
2. if we force create partition, we can call function \"delete_partition_windows\" or \"clean_partition_windows\" before create partition.
3. suggest to delete partition after test. (e.g. get_windows_disks_index ->  configure_empty_disk -> clean_partition_windows